addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiAqaraSensorMagnetHandler.java

```diff
@@ -0,0 +1,97 @@
+package org.openhab.binding.mihome.handler;
+
+import static org.openhab.binding.mihome.XiaomiGatewayBindingConstants.*;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+
+public class XiaomiAqaraSensorMagnetHandler extends XiaomiSensorBaseHandlerWithTimer {
```

Comment:
Why not use the already existing XiaomiSensorMagnetHandler? Seems to me this one does the same, except evalutating the hearbeat message also by status. All the sensors I've seen so far (this one not included), have the problem that they do not state the correct status in their heartbeat -> Window shows open when closed
That's why all the Sensor handlers handle parseReport differently from parseHeartbeat.
I like the refactoring with String literals though!
Perhaps refactor XiaomiSensorMagnetHandler and map it in the XiaomiHandlerFactory to the new aqara sensor as well?